src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -929,14 +929,14 @@ public static CommandElement bool(Text key) {
 
         @Override
         protected Iterable<String> getChoices(CommandSource source) {
-            return Arrays.asList(this.type.getEnumConstants()).stream()
-                .map(input -> input == null ? null : input.name())
+            return Arrays.stream(this.type.getEnumConstants())
+                .map(Enum::name)
                 .collect(Collectors.toList());
         }
 
         @Override
         protected Object getValue(String choice) throws IllegalArgumentException {
-            return Enum.valueOf(this.type, choice.toUpperCase());
+            return Enum.valueOf(this.type, choice);
```

Comment:
This is going to make things fail if user input does not match the enum constant name. A better choice would be to populate a `Map<String, T>` during argument creation, with the `String` key *lowercased*, and simply get the `keySet` for choices, and a simple `get` here.